shotover-proxy/tests/runner/observability_int_tests.rs

```diff
@@ -1,17 +1,77 @@
+        r#"shotover_chain_total{chain="redis_chain"}"#,
+        r#"# TYPE shotover_transform_total counter"#,
+        r#"shotover_transform_total{transform="QueryCounter"}"#,
+        r#"shotover_transform_total{transform="Null"}"#,
+        r#"# TYPE shotover_chain_failures counter"#,
+        r#"shotover_chain_failures{chain="redis_chain"}"#,
+        r#"# TYPE shotover_transform_failures counter"#,
+        r#"shotover_transform_failures{transform="Null"}"#,
+        r#"shotover_transform_failures{transform="QueryCounter"}"#,
+        r#"# TYPE shotover_available_connections gauge"#,
+        r#"shotover_available_connections{source="Redis Source"}"#,
+        r#"# TYPE shotover_transform_latency summary"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.5"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.9"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.95"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.99"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.999"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="1"}"#,
+        r#"shotover_transform_latency_sum{transform="QueryCounter"}"#,
+        r#"shotover_transform_latency_count{transform="QueryCounter"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.5"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.9"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.95"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.99"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.999"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="1"}"#,
+        r#"shotover_transform_latency_sum{transform="Null"}"#,
+        r#"shotover_transform_latency_count{transform="Null"}"#,
+        r#"# TYPE shotover_chain_latency summary"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.5"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.9"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.95"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.99"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.999"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="1"}"#,
+        r#"shotover_chain_latency_sum{chain="redis_chain"}"#,
+        r#"shotover_chain_latency_count{chain="redis_chain"}"#,
+    ];
+
+    // check we get the metrics on startup
+    let mut body = http_request_metrics().await;
+
+    let lines = body.lines().collect::<Vec<&str>>().len();
+
+    for s in &expected {
```

Comment:
should we also do that length check here?

Maybe pull the length check and element check into a helper function?

Reply:
I moved the element check. Couldn't find an obvious way to split out the length check, might become more clear once we add more metrics tests.